pkg/controller/util/endpoint/controller_utils.go

```diff
@@ -164,27 +164,39 @@ func PodChanged(oldPod, newPod *v1.Pod, endpointChanged EndpointsMatch) (bool, b
 		labelsChanged = true
 	}
 
-	// If the pod's deletion timestamp is set, remove endpoint from ready address.
-	if newPod.DeletionTimestamp != oldPod.DeletionTimestamp {
+	// If the pod's deletion timestamp has been set, remove endpoint from ready address.
+	if oldPod.DeletionTimestamp == nil && newPod.DeletionTimestamp != nil {
```

Comment:
just curious, it may happen that the old pod has DeletionTimestamp != nil and the new pod nil?

Reply:
No, that would imply that the pod had been un-deleted.
(FWIW, I only changed this because it meant that otherwise testing deletion had to be the last test in `Test_podChanged`, which was confusing. It could be fixed in other ways. But I believe this change is correct.)

Reply:
(Rob's suggestion to rewrite the test with a modify func ended up making this moot, so I reverted this change.)